### ISSUE TITLE
Problem with apply functions

### ISSUE THREAD

**Issue by alexsiqingliu** [End-to-end-provenance/RDataTracker#102]
ddg.run causes errors when there are apply functions:

```
hightemp <- function(date){
  temp.df <- met.df[met.df$jdate == date, ]
  return(max(temp.df$temp))
}
jdate <- seq(1,365)
sapply(jdate, hightemp)
```

This is the error that I'm seeing:
```
Error in match.call(sys.function(-1), call = call) : 
  ... used in a situation where it does not exist 
5 match.call(sys.function(-1), call = call) 
4 ddg.function() at <text>#3
3 FUN(X[[i]], ...) 
2 lapply(X = X, FUN = FUN, ...) 
1 sapply(jdate, hightemp) 
```

**Comment by MKLau** [comment]
Hey Alex, likes like your met.df object is defined outside of your function. That might be causing the issue you're seeing.

**Comment by blernermhc** [comment]
This is tickling a bug in RDataTracker.  It does not know how to hook in the calls to the functions that are passed to sapply (or other higher-order functions).

**Comment by blernermhc** [comment]
Fixed in 2.23
This no longer results in an error.  We now see the internals of the functions that get called via lapply and other higher-order functions.  We may want to reconsider whether this is what we want the ddg to look like ultimately.
